backend-tests/tests/test_devauth_v2.py

```diff
@@ -1224,3 +1224,134 @@ def compare_aset(authset, api_authset):
+
+        # Create a user for the default tenant
+        default_tenant_user = create_user('root@admin.com', 'foobarbaz', default_tenant.id)
+        r = self.uc.call('POST',
+                    useradm.URL_LOGIN,
+                    auth=(default_tenant_user.name, default_tenant_user.pwd))
+        assert r.status_code == 200
+        default_utoken = r.text
+
+        # Create a device, and try to authorize
+        create_random_authset(default_utoken, default_tenant.tenant_token)
+
+        # Device must show up in the default tenant's account
+        r = self.devauthm.with_auth(default_utoken).call('GET',
+                                            deviceauth_v2.URL_DEVICES)
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 1
+        api_dev = api_devs[0]
+        assert api_dev['status'] == 'pending'
+
+
+    def test_valid_tenant_not_duplicated_for_default_tenant(self, clean_mongo):
+        """Verify that a valid tenant-token login does not show up in the default tenant's account"""
+
+        default_tenant = create_tenant('default-tenant')
+
+        # Restart the deviceauth service with the new token belonging to `default-tenant`
+        deviceauth_cli = CliDeviceauth()
+        deviceauth_cli.add_default_tenant_token(default_tenant.tenant_token)
+
+        # Create a user for the default tenant
+        default_tenant_user = create_user('root@admin.com', 'foobarbaz', default_tenant.id)
+        r = self.uc.call('POST',
+                    useradm.URL_LOGIN,
+                    auth=(default_tenant_user.name, default_tenant_user.pwd))
+        assert r.status_code == 200
+        default_utoken = r.text
+
+        # Create a second user, which has it's own tenant token
+        tenant1 = create_tenant('tenant1')
+
+        tenant1_user = create_user('foo@bar.com', 'foobarbaz', tenant1.id)
+        r = self.uc.call('POST',
+                    useradm.URL_LOGIN,
+                    auth=(tenant1_user.name, tenant1_user.pwd))
+        assert r.status_code == 200
+        tenant1_utoken = r.text
+
+        # Create a device, and try to authorize
+        create_random_authset(tenant1_utoken, tenant1.tenant_token)
+
+        # Device should show up in the 'tenant1's account
+        r = self.devauthm.with_auth(tenant1_utoken).call('GET',
+                                            deviceauth_v2.URL_DEVICES)
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 1
+        api_dev = api_devs[0]
+        assert api_dev['status'] == 'pending'
+
+
+        # Device must not show up in the default tenant's account
+        r = self.devauthm.with_auth(default_utoken).call('GET',
+                                            deviceauth_v2.URL_DEVICES)
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 0
+
+
+    def test_invalid_tenant_token_never_added(self, clean_mongo):
```

Comment:
This was discussed heavily on the backend channel yesterday. We concluded that we will have to support invalid tokens after all, and they will be redirected to the default tenant. Can you change the test case condition?

For now it's going to fail, since I haven't implemented it yet. If you want you can add a `@pytest.mark.xfail(strict=True)` to invert the pass condition while keeping the code correct.